source/helpers/components/compatibility.ts

```diff
@@ -23,24 +23,84 @@ import { liquidEngine } from '../engines';
  * This takes an optional Markdown block (`details`) that should provide more
  * information about the implementation differences or the old behavior.
  */
-export const compatibility = async (
-  details: string,
-  dart: string | boolean | null = null,
-  libsass: string | boolean | null = null,
-  node: string | boolean | null = null,
-  ruby: string | boolean | null = null,
-  feature: string | null = null,
-  useMarkdown = true,
-) =>
-  liquidEngine.renderFile('compatibility', {
+export const compatibility = async (details: string, ...opts: string[]) => {
+  const options = parseCompatibilityOpts(...opts);
+  return liquidEngine.renderFile('compatibility', {
     details: stripIndent(details),
-    dart,
-    libsass,
-    node,
-    ruby,
-    feature,
-    useMarkdown,
+    ...options,
   });
+};
+
+interface CompatibilityOptions {
+  dart: string | boolean | null;
+  libsass: string | boolean | null;
+  node: string | boolean | null;
+  ruby: string | boolean | null;
+  feature: string | null;
+  useMarkdown: boolean;
+}
+
+const extend = <
+  K extends keyof CompatibilityOptions,
+  V extends CompatibilityOptions[K],
+>(
+  value: V,
+  obj: CompatibilityOptions,
+  key: K,
+) => {
+  obj[key] = value;
+};
+
+/**
+ * Take text `inputs` list and convert it into an object of all arguments
+ * suitable for the `compatibility.liquid` template.
+ */
+const parseCompatibilityOpts = (...args: string[]): CompatibilityOptions => {
+  const keyValueRegex = /(\w+):(.*)/;
```

Comment:
Maybe we can blindly split on the colon and let JSON do the heavy lifting:

```javascript
const keyValueRegex = /(.*?):(.*)/
const entries = args.map(arg => {
  const match = arg.match(keyValueRegex)
  if (!match) throw new Error(`Arguments should be in the format 'key:value', got ${arg}.`)
  const key = match[1].trim()
  let value = match[2].trim()
  try {
    value = JSON.parse(value) // handles true, false, null, numbers, strings...
  } catch (e) {
    throw new Error(`Unable to parse argument ${key} with value ${value}. Try wrapping it in double quotes: ${key}:"${value}"`)
  }
  return [key, value]
})
```

This will force us to use the syntax closer to Liquid's implementation, where things like version numbers `1.2.3` and descriptions `some feature` should be quoted to be parsed correctly.

Then we can very easily get a regular object and override the defaults:

```javascript
const kwargs = {...defaults, ...Object.fromEntries(entries)}
```

Reply:
@jerivas Hm, I'm a bit torn on this... I like the cleanliness of your suggestion, but I also hesitate to require double-quoted input args, mostly because I find them annoying to type. 😅 

I also think `JSON.parse('1.0.1')` will throw an error?

Reply:
Yes, it will throw and nudge users towards the double quotes. The "right" way in this case is `'foo:"1.1.1"'`, which will extract `value` as `"1.1.1"` (with the double quotes as part of the match), and JSON will correctly parse it as string.

I don't have strong feelings, so I'm ok with keeping the current implementation, but IMO it's easier to understand "it's JSON after the colon" than "it's a custom mini language where some keywords have special meaning". Plus, it seems easier to maintain and reuse.

Reply:
Not a huge fan of the double quotes either so my vote is for this implementation but I'm very excited either way because this is a major improvement ✨

Reply:
@jerivas Okay, I was convinced. See updated approach: https://github.com/oddbird/sass-site/commit/4a66bbac2bb8b7cc2afc8dd4ee6a0e39abcb569e